### PR TITLE
Add inspect to "Object" interface

### PR DESCRIPTION
**Pull request by mweitzel**
Add inspect method to Object. Call inspect instead of to_s on child elements from from `Hash.to_s` and `Array.to_s`

Reference #698, still need to expose `inspect` on all classes before closing.

### CODE REVIEW

File: vm/decimal.go [hunk 1]
```diff
@@ -689,6 +689,11 @@ func (d *DecimalObject) ToString() string {
 	return fs
 }
 
+// Inspect delegates to ToString
+func (d *DecimalObject) Inspect() string {
```
[the remark this answers is not in the document]
Comment (mweitzel) [reply vm/decimal.go]: There are multiple method in this module with signatures for both `(d *DecimalObject)` and `(f *DecimalObject)`. Ignoring this unless a human directs one way or another.

[the remark this answers is not in the document]
Comment (eliothedeman) [reply vm/decimal.go]: Yeah, we need to clean up our receiver naming scheme. This is the correct way to name receivers btw.

File: vm/hash.go [hunk 2]
```diff
@@ -1206,12 +1206,7 @@ func (h *HashObject) ToString() string {
 	var pairs []string
 
 	for _, key := range h.sortedKeys() {
-		// TODO: Improve this conditional statement
-		if _, isString := h.Pairs[key].(*StringObject); isString {
-			pairs = append(pairs, fmt.Sprintf("%s: \"%s\"", key, h.Pairs[key].ToString()))
-		} else {
-			pairs = append(pairs, fmt.Sprintf("%s: %s", key, h.Pairs[key].ToString()))
-		}
+		pairs = append(pairs, fmt.Sprintf("%s: %s", key, h.Pairs[key].Inspect()))
```
Comment (eliothedeman) [inline vm/hash.go]: 👍

File: vm/string.go [hunk 3]
```diff
@@ -1652,6 +1652,11 @@ func (s *StringObject) ToString() string {
 	return s.value
 }
 
+// Inspect wraps ToString with double quotes
+func (s *StringObject) Inspect() string {
+	return "\"" + s.ToString() + "\""
```
Comment (eliothedeman) [inline vm/string.go]: prefer 
```go
fmt.Sprintf(`"%s"`, s.ToString())
```

**Comment by st0012** [comment]
so far so good 👍

**Comment by hachi8833** [comment]
I think you can merge this ❤️ @st0012 
Then I'd like to try this.

**Comment by st0012** [comment]
@mweitzel thanks for your work ❤️ sorry took so long to merge this

**Pull request merged**